tests/components/pooldose/test_init.py

```diff
@@ -117,3 +117,69 @@ async def test_setup_entry_timeout_error(
     await hass.async_block_till_done()
 
     assert mock_config_entry.state is ConfigEntryState.SETUP_RETRY
+
+
+async def test_migrate_entity_unique_ids(
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test migration of entity unique IDs."""
+    # Add config entry first
+    mock_config_entry.add_to_hass(hass)
```

Comment:
Instead, create a custom mock config entry specifically for this test that has version 1.1